examples/vnet/doca_dpdk.c

```diff
@@ -638,6 +638,68 @@ static void doca_dpdk_build_raw_data(uint8_t **header,
 			layer->fill_data(header, cfg, type);
 	}
 }
+ static int
+ doca_dpdk_modify_encap_action(struct doca_dpdk_action_entry *entry,
+         struct doca_flow_actions *pkt_action)
+ {
+     uint8_t *header;
+	 uint16_t protocol;
+     struct rte_flow_action *action = entry->action;
+     struct doca_dpdk_action_rawencap_data *encap = &entry->action_data.rawencap;
+     struct doca_flow_encap_action *encap_data = &pkt_action->encap;
+
+	header = encap->data;
+	/* ETH */
+	struct rte_ether_hdr eth_hdr;
+	memset(&eth_hdr, 0, sizeof(struct rte_ether_hdr));
+	if (!doca_is_mac_zero(encap_data->src_mac))
+		rte_ether_addr_copy(
+			(const struct rte_ether_addr *)encap_data->src_mac,
+			&eth_hdr.s_addr);
+	if (!doca_is_mac_zero(encap_data->dst_mac))
+		rte_ether_addr_copy(
+			(const struct rte_ether_addr *)encap_data->dst_mac,
+			&eth_hdr.d_addr);
+	protocol = encap_data->src_ip.type == DOCA_IPV4 ?
+		RTE_ETHER_TYPE_IPV4 : RTE_ETHER_TYPE_IPV6;
+	eth_hdr.ether_type = rte_cpu_to_be_16(protocol);
+	memcpy(header, &eth_hdr, sizeof(eth_hdr));
+	header += sizeof(eth_hdr);
+
+	/* IP */
+	struct rte_ipv4_hdr ipv4_hdr;
+	memset(&ipv4_hdr, 0, sizeof(struct rte_ipv4_hdr));
+	if (!doca_is_ip_zero(&encap_data->src_ip))
+		ipv4_hdr.src_addr = encap_data->src_ip.a.ipv4_addr;
+	if (!doca_is_ip_zero(&encap_data->dst_ip))
+		ipv4_hdr.dst_addr = encap_data->dst_ip.a.ipv4_addr;
+		ipv4_hdr.next_proto_id = IPPROTO_UDP;
```

Comment:
why UDP? you need to test the tunnel type to decide if you need udp or not, for example GRE

Reply:
right, i did it only for vxlan. Fixed